### ISSUE TITLE
Channel Manager (CM) Main - Distribution API

### ISSUE THREAD

**Issue by stefanhopman1** [domits1/Domits#440]
## 🚀 Feature Summary

A Channel Manager enables centralized distribution of listings across multiple booking channels from one platform. It synchronizes availability, pricing, content, and reservations in near real time to prevent overbookings. Hosts and operators gain full control, automation, and performance insights across all connected channels (Airbnb, Booking, VRBO, Domits, External Direct Booking Website and more).

----

[PMS](https://github.com/domits1/Domits/issues/149) Capabilities: [Listings](https://github.com/domits1/Domits/issues/2792), [CRS](https://github.com/domits1/Domits/issues/501), [CM](https://github.com/domits1/Domits/issues/440), [Direct Bookings](https://github.com/domits1/Domits/issues/2600), [RMS](https://github.com/domits1/Domits/issues/436), [Calendar & Prices](https://github.com/domits1/Domits/issues/36), [Inbox](https://github.com/domits1/Domits/issues/767), [Tasks](https://github.com/domits1/Domits/issues/2794), [Finance](https://github.com/domits1/Domits/issues/163), [Reviews](https://github.com/domits1/Domits/issues/90), [Intelligence](https://github.com/domits1/Domits/issues/1490), [Settings](https://github.com/domits1/Domits/issues/2793), [Integrations](https://github.com/domits1/Domits/issues/433), [Agentic AI](https://github.com/domits1/Domits/issues/442)
Use Case / Scenario Testing: [PMS](https://github.com/domits1/Domits/issues/2262), [Listings](https://github.com/domits1/Domits/issues/3039), [CRS](https://github.com/domits1/Domits/issues/2265), [CM](https://github.com/domits1/Domits/issues/2266), [Direct Bookings](https://github.com/domits1/Domits/issues/3038), [RMS](https://github.com/domits1/Domits/issues/2267), [Calendar & Prices](https://github.com/domits1/Domits/issues/2299), [Tasks](https://github.com/domits1/Domits/issues/3040), [Inbox](https://github.com/domits1/Domits/issues/2291), [Finance](https://github.com/domits1/Domits/issues/2287)

----

<img width="1306" height="517" alt="Image" src="https://github.com/user-attachments/assets/c11849d3-fd93-4f91-b301-f6169aec16a4" />

**- First phase**

- Understand Channel Management: https://bookdomits.com/channel-management-distribution-hospitality/
- [x] Channel Management: #619

**- iCal for import/export sync**

- [x] https://github.com/domits1/Domits/issues/587

**- External Direct Reservations as Distribution Channels**
- [x] #2600 

**Distribution API Requirements for partners**
- [ ] #2192
- [ ] #2861 
- [ ] #2869

**- Dutch/European Distribution Channels**
- [ ] #2374
- [ ] #2724 
- [ ] #2596
- [ ] #1572
- [ ] #1573
- [ ] Channel Management (Europe): HomeToGo Integration Overview, Management, Requirements

**- Global Distribution Channels**
- [ ] #1226
- [ ] #1561
- [ ] #1571
- [ ] #1901
- [ ] Channel Management: Integration with VRBO, Hopper, Trip.com, Agoda and requirements
- [ ] Channel Management: Integration with luxury channels (OneFineStay, Oliver’s Travels, Plum Guide, Luxury Escapes, StayOne)
- [ ] Channel Management: Integration with expanded channels (niche websites/smaller OTA's)

**- Global Meta Search**
- [ ] #2240

Partners
- [ ] #2599 
- [ ] VRBO Elite Partner
- [ ] #2618
- [ ] Google Vacation Rentals Partner
- [ ] Expedia Partner
- [ ] #2619
- [ ] Trip.com Partner

Docs
- [ ] https://github.com/domits1/Domits/issues/2225

----

**- Later**
- [ ] #36 View different locations, types of properties and search for availability or do bulk updates.
- [ ] #767 
- [ ] Multi-unit support. Connect multiple room types to single properties and you can assign guests to units.
- [ ] Cross-listings. Flexible arrangements for big groups. Combine multiple listings into one with no chances of double-bookings, across all major channels.
- [ ] Contact information: Add contact information of the person in charge of the channel
- [ ] https://github.com/domits1/Domits/issues/588

Resources: 
https://www.altexsoft.com/blog/vacation-rental-channel-managers/

PMS with channel manager:
- [ ] https://www.guesty.com/features/channel-manager/
- [ ] https://www.cloudbeds.com/channels/ 
- [ ] https://www.hostaway.com/features/channel-manager/
- [ ] https://www.lodgify.com/vacation-rental-channel-manager/
- [ ] https://apaleo.com/open-apis
- [ ] https://www.siteminder.com/integrations/
- [ ] https://www.mews.com/en/products/marketplace

Channel Management for Distribution examples:
- [ ] https://docs.channex.io/
- [ ] [NextPax](https://nextpax.com/) (Dutch)
- [ ] [Maxxton](https://www.maxxton.com/solutions/channel-manager/) (Dutch)

OTA Examples: 
- [ ] Booking, Expedia, Airbnb

See more: https://skift.com/skift-short-term-rental-250

----

## 🧩 Problem Statement
What problem does this feature solve? Who is affected, and why does it matter? > Focus on the problem, not the solution.

No centralized channel management for distribution. No more double bookings.

---

## 🎯 Desired Outcome
What does success look like if this feature is implemented?

Centralized channel management for distribution.

---

## 👥 Target Users
Who benefits from this feature?
- [ ] Guests / Customers
- [x] Hosts / Property Managers / Enterprise
- [ ] Admin / Internal users
- [ ] Developers
- [ ] Other: _______

---

## 📈 Business & User Impact
How valuable is this feature?

- **Impact level:**
  - [ ] Low (nice-to-have)
  - [ ] Medium (improves workflows)
  - [ ] High (unblocks key use cases)
  - [x] Critical (strategic / revenue-impacting)

- **User reach:**
  - [ ] Few users
  - [ ] Many users
  - [x] All users
  - [ ] Unknown

---

## ⚠️ Constraints & Considerations
- Performance
- Security
- Privacy
- Accessibility
- Backward compatibility
- External dependencies

---

## 🧪 Acceptance Criteria / User Stories
How can we validate that this feature is complete?

- [ ] Given X, when Y, then Z

----
👥 User Roles

- Host / Property Manager
- Channel Manager / Revenue Manager
- Admin / System Operator
- Cleaner / Maintenance Staff
- Accountant / Finance Manager
- OTA System (external API partner)

🏘️ 1. Property Listing Management

- [ ] As a host, I want to connect my properties to multiple OTAs (Airbnb, Booking.com, VRBO) so I can manage them centrally.
- [ ] As a host, I want to import existing listings from OTAs into the system to avoid duplicate data entry.
- [ ] As a host, I want to push property details (photos, descriptions, amenities) to all connected channels.
- [ ] As a host, I want to update listing information (description, amenities, photos) once and sync it everywhere.
- [ ] As a host, I want to unpublish a property from specific channels temporarily.
- [ ] As a host, I want to group properties by portfolio, city, or client for easier management.
- [ ] As a host, I want to view which channels each property is connected to.
- [ ] As an admin, I want to validate property data before pushing it to external APIs.

📅 2. Availability & Calendar Synchronization

- [ ] As a host, I want my property calendar to sync across all OTAs automatically.
- [ ] As a host, I want bookings made on any channel to instantly block dates on all others to prevent double bookings.
- [ ] As a host, I want to manually block specific dates across all channels.
- [ ] As a host, I want to define booking rules (minimum stay, maximum stay, lead time) synced to OTAs.
- [ ] As a system, I want to detect and resolve calendar conflicts automatically.
- [ ] As a host, I want to view a unified multi-property calendar in one dashboard.
- [ ] As an admin, I want to monitor synchronization logs and error messages.
- [ ] As a system, I want to sync data via both iCal and API connections for full coverage.

💵 3. Rates, Pricing & Promotions

- [ ] As a host, I want to update nightly rates centrally and push them to all OTAs.
- [ ] As a revenue manager, I want to set rate plans (standard, refundable, non-refundable) for each channel.
- [ ] As a host, I want to create discounts and promotions and distribute them to selected channels.
- [ ] As a system, I want to automatically convert currencies based on each OTA’s requirements.
- [ ] As a host, I want to apply markup or markdown percentages per channel to adjust for commissions.
- [ ] As a host, I want to schedule future price changes automatically.
- [ ] As a revenue manager, I want to view rate parity reports to ensure consistent pricing.
- [ ] As an admin, I want to restrict unauthorized users from editing rates.

📦 4. Booking Management

- [ ] As a host, I want all bookings from every channel to appear in one centralized dashboard.
- [ ] As a host, I want to view booking details (guest info, dates, price, channel source).
- [ ] As a host, I want to modify or cancel bookings and push updates to OTAs.
- [ ] As a system, I want to automatically update availability after a booking or cancellation.
- [ ] As a host, I want to differentiate between confirmed, pending, and canceled bookings.
- [ ] As a host, I want to export booking data for accounting or CRM use.
- [ ] As an admin, I want to track booking volume by channel.
- [ ] As a system, I want to detect duplicate reservations and flag them.

🔄 5. Channel Connectivity & Integrations

- [ ] As an admin, I want to manage API keys and credentials for each connected OTA.
- [ ] As a system, I want to handle authentication and refresh tokens securely.
- [ ] As a channel manager, I want to configure channel-specific settings (e.g., fees, commissions).
- [ ] As a system, I want to schedule periodic synchronization jobs for rates, availability, and content.
- [ ] As an admin, I want to monitor API health and connection uptime.
- [ ] As a host, I want to reconnect a channel if a connection is lost.
- [ ] As a system, I want to support multiple connection types (API, iCal, XML).
- [ ] As an admin, I want to see version history and connection logs.

🧾 6. Reporting & Analytics

- [ ] As a host, I want to view booking performance by channel.
- [ ] As a revenue manager, I want to see occupancy rates, ADR, and RevPAR by channel and property.
- [ ] As a host, I want to analyze revenue distribution across channels.
- [ ] As an admin, I want to track connection errors or sync failures.
- [ ] As a revenue manager, I want a dashboard showing the top-performing channels.
- [ ] As an analyst, I want to export channel performance reports to Excel.
- [ ] As a host, I want to compare current performance vs last year’s.
- [ ] As an admin, I want to generate monthly channel reconciliation reports.

🧩 7. Multi-Property & Multi-Account Management

- [ ] As a host, I want to manage multiple OTA accounts from a single platform.
- [ ] As a manager, I want to assign specific properties to team members.
- [ ] As a management company, I want to handle multiple owners’ properties securely.
- [ ] As an admin, I want to define permission levels per user.
- [ ] As a host, I want to delegate booking approval rights to staff.
- [ ] As an admin, I want to manage organization-level settings for all connected properties.

🔔 8. Notifications & Alerts

- [ ] As a host, I want notifications when new bookings arrive from any channel.
- [ ] As a host, I want alerts for sync failures or connection timeouts.
- [ ] As an admin, I want system error alerts for failed API calls.
- [ ] As a host, I want a daily summary of bookings and cancellations.
- [ ] As a host, I want reminders when OTA credentials are about to expire.
- [ ] As an admin, I want alerts when a channel integration version changes.

💬 9. Guest Communication (optional module)

- [ ] As a host, I want to receive guest messages from all channels in one inbox.
- [ ] As a host, I want to reply to guests directly from the platform.
- [ ] As a host, I want to use automated pre-arrival and checkout messages.
- [ ] As a host, I want to attach check-in instructions and send them automatically.
- [ ] As an admin, I want to monitor message delivery logs.

🔐 10. User Access & Security

- [ ] As an admin, I want to manage users, roles, and access levels.
- [ ] As a user, I want to log in using secure OAuth or SSO.
- [ ] As a host, I want two-factor authentication for my account.
- [ ] As an admin, I want to audit user actions like rate changes and connection edits.
- [ ] As a system, I want to encrypt stored API credentials.
- [ ] As a system, I want to anonymize guest data for GDPR compliance.

⚙️ 11. Automation Rules & Smart Workflows

- [ ] As a host, I want automatic blocking of dates when maintenance is scheduled.
- [ ] As a revenue manager, I want dynamic rate updates based on occupancy rules.
- [ ] As a host, I want automatic closing of listings when occupancy exceeds a threshold.
- [ ] As an admin, I want configurable automation templates (rate sync, calendar updates).
- [ ] As a system, I want to retry failed syncs automatically.

📈 12. Performance Optimization & Monitoring

- [ ] As an admin, I want real-time dashboards showing system performance metrics.
- [ ] As a system, I want to queue synchronization jobs to handle large volumes efficiently.
- [ ] As an admin, I want alerts when synchronization latency exceeds thresholds.
- [ ] As a system, I want to maintain API rate-limit compliance with OTAs.
- [ ] As a system, I want to record the duration of each sync job for performance tuning.

🧮 13. Accounting & Reconciliation

- [ ] As a finance manager, I want to view payouts and commissions per channel.
- [ ] As a host, I want to reconcile OTA invoices against internal booking records.
- [ ] As an admin, I want to apply different commission percentages per channel.
- [ ] As a finance manager, I want to export financial data to my accounting system.
- [ ] As a host, I want to view pending payouts and payment status by channel.

🌍 14. Localization & Currency

- [ ] As a host, I want to view pricing in my local currency.
- [ ] As a system, I want to automatically convert currencies based on current exchange rates.
- [ ] As a host, I want to manage time zones for each property.
- [ ] As a user, I want to view the dashboard in multiple languages.

🧾 15. Audit & Compliance

- [ ] As an admin, I want to keep logs of all channel updates and rate changes.
- [ ] As an admin, I want to view detailed synchronization histories.
- [ ] As a system, I want to store audit trails for 90 days for compliance.
- [ ] As a user, I want to download audit logs for review.

**Comment by Timmhart** [comment]
## User story
As a host, I want to manage my accommodation listings across multiple distribution platforms from a single dashboard, so I can ensure consistent availability, pricing, and updates without manual effort.

## Example mapping

- What if the host wants to add a new distribution channel?
  - The host should be able to easily connect new platforms by entering credentials or API keys. The system should verify the connection and start synchronizing the listings immediately.
- What if there is a synchronization error between the platform and a distribution channel?
  - The system should notify the host of the error and provide troubleshooting steps (e.g., retry syncing or contact support).
  - The error message should include details about the issue (e.g., connection timeout, incorrect credentials) and suggest a solution.
- What if the host wants to update pricing or availability?
  - Any updates made on the host’s dashboard should instantly sync with all connected channels, ensuring consistency across platforms.
  - The host should be notified when the synchronization is complete or if there are any issues.
- What if a booking is made on one of the distribution channels?
  - The system should automatically update availability on all other channels to prevent double-bookings.
  - The host should receive a notification, and the booking should be visible on the host’s dashboard, with details about which platform it originated from.
- What if the host wants to prioritize bookings from a specific channel?
  - The system should allow the host to set a preference for certain channels (e.g., direct bookings on their own website), with options like applying different pricing or promoting specific availability on those channels first.
- What if the host decides to stop using a particular distribution channel?
  - The host should be able to easily disconnect the platform from their dashboard. The system should inform them of any pending bookings that may still need to be managed before disconnecting.
- What if there are inconsistent details between platforms (e.g., different pricing)?
  - The system should highlight discrepancies and allow the host to choose whether to synchronize details across all channels or manually adjust specific listings on certain platforms.
- What if the host wants a report on performance across channels?
  - The system should offer a report summarizing bookings, revenue, and platform performance, with the ability to filter by platform, date range, and other criteria.

## Accept criteria

- [ ] The host should be able to link their accommodation to various distribution channels (e.g., Airbnb, Booking.com, Expedia) from the platform.
- [ ] The system should synchronize availability, pricing, and updates across all connected platforms in real-time.
- [ ] The host should be able to view which platforms their accommodation is listed on, with the option to add or remove channels easily.
- [ ] The platform should provide a clear status update of synchronization for each channel (e.g., success, error).
- [ ] If an error occurs during synchronization, the system should notify the host and provide a way to resolve the issue (e.g., re-sync button or contact support).
- [ ] Changes made to the accommodation details (e.g., pricing, availability) on the host’s platform should automatically update across all linked channels.
- [ ] The host should be able to see a summary of bookings from all channels, with the ability to filter by platform, date, or status.
- [ ] The platform should allow hosts to prioritize certain channels, for example, to favor direct bookings on their own site over third-party platforms.
- [ ] The system should automatically block dates across all platforms when a booking is made on any channel to avoid double-bookings.

**Comment by Bekoo023** [comment]
https://to-do.microsoft.com/sharing?InvitationToken=iqps8OaxUXI1KF4rmykxaQbsBTJXIm0GH9TEiEvfkjslr8x_2CJM5MVnfPMtOE_-s 

ToDo list for Holidu Req
